actionpack/lib/action_dispatch/journey/path/pattern.rb

```diff
@@ -83,7 +83,7 @@ def initialize(separator, matchers)
           end
 
           def accept(node)
-            %r{\A#{visit node}\Z}
+            RegexpCache.anchored(visit(node))
```

Comment:
Here it's much less clear. I count `4771` regexp, `3573` different ones, and the most duplicated one is seen 8 times, and only `698` of them are only seen once.

So not sure if it's really worth it.

Reply:
Yeah, it makes sense these would be mostly all different. I think `visit node` here returns a regex source (i.e. a string) for the whole path. So there we'd get one regexp for each route.

We've got `4407` regexp (and `4407` routes), `3720` different ones, and `3220` of them only appear once. One of them appears 10 times.

So not really worth it for us, although it seems like you've got more routes with the same regex than we do.